core/src/main/java/org/apache/iceberg/BaseEntriesTable.java

```diff
@@ -59,6 +63,16 @@ public Schema schema() {
     return TypeUtil.join(schema, MetricsUtil.readableMetricsSchema(table().schema(), schema));
   }
 
+  @Override
+  public PartitionSpec spec() {
+    return specs.get(defaultSpecId);
```

Comment:
A bit concerned here. Should the partition spec of metadata be equal to the table? From my understanding, the metadata tables should be definitive new tables. Just like we have specified schema for the metadata table.

Reply:
> Should the partition spec of metadata be equal to the table
I think there's method by @szehon-ho which did identity transform of each columns found in given partition spec. It was meant for predicate pushdown in metadata column in #2926 

> the metadata tables should be definitive new tables. Just like we have specified schema for the metadata table
I am not sure if I understand you, to populate data in both data tables and metadata tables, the similar (not exact set of ) underlying files needs to be scanned and reporting different representation of same date. Let's say for PartitionTable, we count number of files and regard together with other columns aggregated on partition level. So from what I can tell, the metadata partitionSpec are only useful to provide predicate pushdown so it can leverage the existing tools when we used to read its original data tables.

Data and metadata tables do have different schemas but identity transformation of original data partition spec to be used in metadata table seem to be a good option to me. Let me know what do you think?

Reply:
Yeah, I see the spec is used only for push-down in this PR. The problem is that the spec is built with the table schema which is not the same as the metadata table schema. This could be a little confusing for the people who used the Java API. Could we just use the spec to do the push-down while not exposing it for the metadata table? Not sure whether there are some better choices.